R/time_event.R

```diff
@@ -99,32 +106,39 @@ prep.step_time_event <- function(x, training, info = NULL, ...) {
     trained = TRUE,
     rules = x$rules,
     columns = col_names,
+    keep_original_cols = get_keep_original_cols(x),
     skip = x$skip,
     id = x$id
   )
 }
 
 #' @export
 bake.step_time_event <- function(object, new_data, ...) {
-  if (length(object$column) == 0L) {
-    # Empty selection
-    return(new_data)
-  }
+  col_names <- names(object$columns)
+  check_new_data(col_names, object, new_data)
+
+  for (col_name in col_names){
+    tmp <- get_time_events(
+      rules = object$rules,
+      column = object$columns[col_name],
+      name = col_name,
+      new_data = new_data
+    )
 
-  new_cols <- purrr::imap_dfc(object$columns, time_event_helper,
-                              new_data, object$rules)
+    names(tmp) <- paste(col_name, names(tmp), sep = "_")
+    tmp[] <- lapply(X = tmp, FUN = vctrs::vec_cast, integer())
```

Comment:
What does the `[]` do after `tmp` here? is is needed?

Reply:
Without the `[]` the result of `lapply` will be a list and you'd have to do another `as_tibble` call. `df[] <- lapply(df,...)` is a pretty common base R pattern for iterating over columns and getting a data frame back, in my experience. If you think it's too cryptic I can just do

```
tmp <- lapply(X = tmp,...)
tmp <- as_tibble(tmp)
```

Happy to do whichever...

Reply:
looks good. Keep it!